controllers/core/event_controller.go

```diff
@@ -0,0 +1,207 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package controllers
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/config"
+	"github.com/aws/amazon-vpc-resource-controller-k8s/pkg/k8s"
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	MaxEventConcurrentReconciles = 4
+	EventCreatedPastMinutes      = 2
+	EventFilterKey               = "reportingComponent"
+)
+
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;update;patch;list;watch
+
+type EventReconciler struct {
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+	K8sAPI k8s.K8sWrapper
+}
+
+var (
+	eventControllerTotalEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "total_events_reconciled_by_event_controller",
+			Help: "The number of events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerSGPEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "security_group_pod_events_reconciled_by_event_controller",
+			Help: "The number of SGP events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerCNEventsCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "custom_networking_events_reconciled_by_event_controller",
+			Help: "The number of custom networking events that were reconcile by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventControllerEventFailureCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_get_events__by_event_controller",
+			Help: "The number of failures on getting events by the controller",
+		},
+		[]string{"operation"},
+	)
+
+	eventReconcileOperationLatency = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "event_reconcile_operation_latency",
+			Help: "Event controller reconcile events latency in ms",
+		},
+		[]string{"time"},
+	)
+
+	prometheusRegistered = false
+	reconcileEvents      = "reconcile_events"
+)
+
+func PrometheusRegister() {
+	if !prometheusRegistered {
+		metrics.Registry.MustRegister(eventControllerTotalEventsCount)
+		metrics.Registry.MustRegister(eventControllerSGPEventsCount)
+		metrics.Registry.MustRegister(eventControllerEventFailureCount)
+		metrics.Registry.MustRegister(eventControllerCNEventsCount)
+		metrics.Registry.MustRegister(eventReconcileOperationLatency)
+		prometheusRegistered = true
+	}
+}
+
+func (r *EventReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Since the node will [stay as managed](https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/4cfca50bbfff4410386d1565f5c78195ac34272b/pkg/node/manager/manager.go#L225-L237), we will only check if their subnet has changed.
https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/master/pkg/node/manager/manager.go#L202-L210